src/main/java/edu/pdx/cs/Client.java

```diff
@@ -49,6 +51,72 @@ void setLogfile(Path logpath) {
         this.logpath = logpath;
     }
 
+    /**
+     * Lists directories w/in the current working directory on remote server
+     * @return an array of <code>FTPFile</code> objects
+     */
+    protected FTPFile[] listRemoteDirectories() throws IOException {
+        String path = ftp.printWorkingDirectory();
+        return ftp.listDirectories(path);
+    }
+
+    /**
+     * Lists files w/in the current directory on remote server
+     * @return an array of <code>FTPFile</code> objects
+     */
+    protected FTPFile[] listRemoteFiles() throws IOException{
+        String path = ftp.printWorkingDirectory();
+
+        FTPFile[] files =  ftp.listFiles(path, new FTPFileFilter() {
+            @Override
+            public boolean accept(FTPFile ftpFile) {
+                return !ftpFile.isDirectory();
+            }
+        });
+
+        return files;
+    }
+
+    /**
+     * Converts names of <code>FTPFile</code> objects to strings
+     * @param files array of <code>FTPFiles</code> objects
+     * @return <code>ArrayList</code> of names in <code>String</code> format
+     */
+    protected ArrayList<String> fileDirectoryListStrings(FTPFile[] files){
+       ArrayList<String> names = null;
+
+       for(FTPFile file:files)
+           names.add(file.getName());
+
+       return names;
+    }
```

Comment:
As far as I can tell, this is dead code

Reply:
My intention was for main() to get access to the FTPFile arrays returned by listRemoteDirectories() and listRemoteFiles().  fileDirectoryListStrings() can then be used to get the name of each FTPFile object as a string for printing. i.e.

FTPFile[] directories = client.listRemoteDirectories();
ArrayList<String> dirNames = fileDirectoryListStrings(directories);
for(name:dirNames)
   System.out.print(name + "   ");

If this function is redundant, main() could also simply use obj.getName().  Happy to get rid of that function if .getName() seems like the better way to go.